File: src/types/task.ts
import type { Action } from "./action"

interface ITask{
    id?:number,
    name:string,
    completed:boolean,
    parent_id?:string,
    desc:string,
    actions:Action[]
    createdAt:Date,
    dueDate:Date
}
interface ITaskView{
    id?:number,
    name:string,
    desc:string,
    parent_id?:string,
    completed:boolean,
    actions:Action[]
    createdAt:Date,
    dueDate:Date
    avatar?:string
}

export type Task = ITask;
export type TaskView = ITaskView;

export type TaskList = Array<Task>;

export type TaskStatus = "all" | "completed" | "incomplete";

File: src-tauri/src/schema/task.rs
use serde::{Deserialize,Serialize};

#[derive(Deserialize,Serialize, Debug)]
pub struct TaskRecord {
    pub id: String,
    pub completed: bool,
    pub parent_id: Option<String>,
    pub name: String,
    pub desc: String,
    pub actions: Vec<String>,
    pub created_at: String,
    pub start_at: Option<String>,
}

#[derive(Deserialize,Serialize, Debug)]
pub struct TaskView {
    pub id: String,
    pub name: String,
    pub desc: String,
    pub actions: Vec<String>,
    pub children: Vec<TaskView>,
}